src/Microsoft.Health.Dicom.Core/Extensions/DicomElementExtensions.cs

```diff
@@ -0,0 +1,146 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Dicom;
+using EnsureThat;
+using Microsoft.Health.Dicom.Core.Features.Validation;
+
+namespace Microsoft.Health.Dicom.Core.Extensions
+{
+    /// <summary>
+    /// Extension methods for <see cref="DicomTag"/>.
+    /// </summary>
+    public static class DicomElementExtensions
+    {
+        private static readonly IReadOnlyDictionary<DicomVR, Func<DicomElement, object>> Readers = new Dictionary<DicomVR, Func<DicomElement, object>>()
+        {
+            { DicomVR.AE, GetStringValue },
+            { DicomVR.AS, GetStringValue },
+            { DicomVR.AT, GetUIntValue },
+            { DicomVR.CS, GetStringValue },
+            { DicomVR.DA, GetDAValue },
+            { DicomVR.DS, GetStringValue },
+            { DicomVR.DT, GetDTValue },
+            { DicomVR.FL, GetFloatValue },
+            { DicomVR.FD, GetDoubleValue },
+            { DicomVR.IS, GetStringValue },
+            { DicomVR.LO, GetStringValue },
+            { DicomVR.PN, GetStringValue },
+            { DicomVR.SH, GetStringValue },
+            { DicomVR.SL, GetIntValue },
+            { DicomVR.SS, GetShortValue },
+            { DicomVR.TM, GetTMValue },
+            { DicomVR.UI, GetStringValue },
+            { DicomVR.UL, GetUIntValue },
+            { DicomVR.US, GetUShortValue },
+        };
+
+        public static object GetSingleValue(this DicomElement element)
+        {
+            EnsureArg.IsNotNull(element, nameof(element));
+            if (element.Count != 1)
+            {
+                return null;
+            }
+
+            if (!Readers.ContainsKey(element.ValueRepresentation))
+            {
+                Debug.Fail($"Need reader for VR {element.ValueRepresentation}");
+                return null;
+            }
+
+            return Readers[element.ValueRepresentation].Invoke(element);
+        }
+
+        private static object GetDAValue(DicomElement element)
+        {
+            string stringDate = (string)SafeGetValue<string>(element);
+
+            if (stringDate == null ||
+                !DicomElementMinimumValidation.TryParseDA(stringDate, out DateTime result))
+            {
+                return null;
+            }
+
+            return result;
+        }
+
+        private static object SafeGetValue<T>(DicomElement element)
+        {
+            try
+            {
+                return element.Get<T>();
+            }
+            catch
+            {
+                return null;
```

Comment:
add log

Reply:
how to add log for static methods?

Reply:
make it instance class?

Reply:
You could add the option of passing a logger

Reply:
Or you could have a statically-scoped `ILoggerFactory<T>` that was configured in start-up that this class can use. Given this is a library, it is a bit more difficult to do that though.

Reply:
However, should we really be swallowing type-mismatch exceptions like this?

Reply:
This class is removed